### PR TITLE
Fixes in hypermedia client for thumbnails

### PR DESCRIPTION
**Pull request by clemencetbk**

### CODE REVIEW

File: core/d2l-content-viewer/d2l-content-viewer.js [hunk 1]
```diff
@@ -192,10 +192,12 @@ class ContentViewer extends InternalLocalizeMixin(LitElement) {
 	}
 
 	async _loadThumbnails() {
-		const result = this.activity ? await this.hmClient.getThumbnails(this._resourceEntity)
-			: await this.client.getThumbnails();
-		if (result) {
-			this._thumbnails = result.Value;
+		if (this.activity) {
```
Comment (MichaelKovalchuk) [inline core/d2l-content-viewer/d2l-content-viewer.js:195]: any issues with the code before? i am not sure what changed logically here

File: core/d2l-content-viewer/src/clients/hypermedia-client.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ export default class HypermediaClient {
 			const metadataEntity = SirenParse(metadataResponse);
 			return metadataEntity.properties;
 		} catch (e) {
-			if (e.message !== 'Not Found') throw e;
+			if (e.code != 404) throw e;
```
Comment (MichaelKovalchuk) [inline core/d2l-content-viewer/src/clients/hypermedia-client.js:76]: !== as != can give you some unpredictable results

File: core/d2l-content-viewer/d2l-content-viewer.js (hunk 1 above)
Comment (clemencetbk) [reply core/d2l-content-viewer/d2l-content-viewer.js:195]:
```   
if (this.activity) {
			const result = await this.hmClient.getThumbnails(this._resourceEntity);
			if (result) this._thumbnails = result.value;
		} else {
			const result = await this.client.getThumbnails();
			if (result) this._thumbnails = result.Value;
		 }
```

the hypermedia route returns a result with attribute "value" while  the REST route returns a result with attribute "Value". Seems like there's a convention to use uppercase attributes for REST and lowercase for hypermedia

**Review by MichaelKovalchuk — APPROVED** [review]

**Comment by clemencetbk** [comment]
image
This check is failing. I think maybe I need to bump eslint to 7?

**Comment by rcausey** [comment]
> image This check is failing. I think maybe I need to bump eslint to 7?

yeah that's what I'm thinking.

**Pull request merged**